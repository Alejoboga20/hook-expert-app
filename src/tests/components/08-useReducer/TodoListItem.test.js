import { shallow } from 'enzyme';
import React from 'react';
import TodoListItem from '../../../components/08-useReducer/TodoListItem';
import { demoTodos } from '../../fixtures/demoTodos';

describe('TodoListItem tests', () => {
  const handleDelete = jest.fn();
  const handleToggle = jest.fn();
  let wrapper = shallow(
    <TodoListItem
      todo={demoTodos[0]}
      index={0}
      handleDelete={handleDelete}
      handleToggle={handleToggle}
    />
  );

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(
      <TodoListItem
        todo={demoTodos[0]}
        index={0}
        handleDelete={handleDelete}
        handleToggle={handleToggle}
      />
    );
  });

  test('should display properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should call delete function', () => {
    wrapper.find('button').simulate('click');
    expect(handleDelete).toHaveBeenCalled();
    expect(handleDelete).toHaveBeenCalledWith(demoTodos[0].id);
  });

  test('should call toggle function', () => {
    wrapper.find('p').simulate('click');
    expect(handleToggle).toHaveBeenCalled();
    expect(handleToggle).toHaveBeenCalledWith(demoTodos[0].id);
  });

  test('should show text properly', () => {
    const p = wrapper.find('p');
    expect(p.text()).toBe(`1 ${demoTodos[0].description}`);
    expect(wrapper.find('p').hasClass('complete')).toBe(false);
  });
});
