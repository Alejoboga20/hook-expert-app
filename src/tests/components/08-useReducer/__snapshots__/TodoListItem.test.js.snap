// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoListItem tests should display properly 1`] = `
<li
  className="list-group-item"
>
  <p
    className="false"
    onClick={[Function]}
  >
    1
     
    Learn React
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Delete
  </button>
</li>
`;
